File: microservice-movie-management/src/main/java/com/claudiu/microservicemoviemanagement/service/MovieService.java
package com.claudiu.microservicemoviemanagement.service;

import com.claudiu.microservicemoviemanagement.model.Movie;
import com.claudiu.microservicemoviemanagement.model.Rating;

import java.util.List;

public interface MovieService {
    List<Movie> allMovies();

    Movie findMovieById(Long id);

    List<Rating> findUserRatings(Long userId);

    List<Rating> findRatingsOfMovie(Long movieId);

    Rating saveRating(Rating rating);

    Movie updateMovie(Movie movie);
}

File: client-service/src/app/services/movie-service/movie.service.ts
import { Injectable } from '@angular/core';
import {HttpClient, HttpHeaders} from '@angular/common/http';
import {Observable} from 'rxjs';
import { Rating } from 'src/app/model/rating';

let API_URL = "http://localhost:8765/api/movie/service/";

@Injectable({
  providedIn: 'root'
})
export class MovieService {

  constructor(private http: HttpClient) { }

  rate(rating: Rating): Observable<any>{
    return this.http.post(API_URL + 'rate',JSON.stringify(rating),
    {headers: {"Content-Type":"application/json; charset=UTF-8"}});
  }

  findAllMovies(): Observable<any>{
    return this.http.get(API_URL + "all",
    {headers: {"Content-Type":"application/json; charset=UTF-8"}});
  }

  findRatingsOfUser(userId: number): Observable<any> {
    return this.http.get(API_URL + "user/" + userId, 
    {headers: {"Content-Type":"application/json; charset=UTF-8"}});
  }

  findUsersOfMovie(movieId: number): Observable<any> {
    return this.http.get(API_URL + "movie/" + movieId, 
    {headers: {"Content-Type":"application/json; charset=UTF-8"}});
  }
}
